### ISSUE TITLE
Better Junit formatter output

### ISSUE THREAD

**Issue by jpserra** [rubocop/rubocop#9780]
## Is your feature request related to a problem? Please describe.

When using junit formatter the output is:
```xml
<?xml version='1.0'?>
<testsuites>
  <testsuite name='rubocop'>
  ...
  </testsuite>
</testsuites>
```

which doesn't include some of the common attributes of junit (as per the [documentation](https://www.ibm.com/docs/en/developer-for-zos/14.1.0?topic=formats-junit-xml-format)) such as:
```
    tests: The number of rules in the provider that were applied.
    failures: The number of rule violations in the provider.
```

## Describe the solution you'd like

Provide a more informative output including the number of total tests ran and the number of failures in the testsuite element.

## Describe alternatives you've considered

alternative 1
```xml
<?xml version='1.0'?>
<testsuites tests='175824' failures='0'>
  <testsuite name='rubocop'>
  ...
  </testsuite>
</testsuites>
```

alternative2
```xml
<?xml version='1.0'?>
<testsuite name='rubocop' tests='175824' failures='0'>
  ...
</testsuite>
```
